play-services-core/src/main/kotlin/org/microg/gms/auth/signin/AuthSignInActivity.kt

```diff
@@ -197,18 +200,37 @@ class AuthSignInActivity : AppCompatActivity() {
 
     private fun finishResult(statusCode: Int, message: String? = null, account: Account? = null, googleSignInAccount: GoogleSignInAccount? = null) {
         val data = Intent()
-        if (statusCode != CommonStatusCodes.SUCCESS) data.putExtra("errorCode", statusCode)
-        data.putExtra("googleSignInStatus", Status(statusCode, message))
-        data.putExtra("googleSignInAccount", googleSignInAccount)
+        if (statusCode != CommonStatusCodes.SUCCESS) data.putExtra(AuthConstants.ERROR_CODE, statusCode)
+        data.putExtra(AuthConstants.GOOGLE_SIGN_IN_STATUS, Status(statusCode, message))
+        data.putExtra(AuthConstants.GOOGLE_SIGN_IN_ACCOUNT, googleSignInAccount)
+        val bundle = Bundle()
         if (googleSignInAccount != null) {
-            data.putExtra("signInAccount", SignInAccount().apply {
+            SignInAccount().apply {
                 email = googleSignInAccount.email ?: account?.name
                 this.googleSignInAccount = googleSignInAccount
-                userId = googleSignInAccount.id ?: getSystemService<AccountManager>()?.getUserData(account, "GoogleUserId")
-            })
+                userId = googleSignInAccount.id ?: getSystemService<AccountManager>()?.getUserData(
+                    account,
+                    AuthConstants.GOOGLE_USER_ID
+                )
+            }.let {
+                data.putExtra(AuthConstants.SIGN_IN_ACCOUNT, it)
+            }
+            SignInCredential().apply {
+                email = googleSignInAccount.email
+                displayName = googleSignInAccount.displayName
+                familyName = googleSignInAccount.familyName
+                givenName = googleSignInAccount.givenName
+                idToken = googleSignInAccount.idToken
+            }.let {
+                bundle.putByteArray(AuthConstants.SIGN_IN_CREDENTIAL, it.toByteArray())
+                bundle.putByteArray(AuthConstants.STATUS, Status.SUCCESS.toByteArray())
```

Comment:
This line does not even depend on the `SignInCredential` object you just created, so it makes little sense to have it in the same `let` block.

Reply:
it changed